### ISSUE TITLE
Pulling of an image does not refresh Images after first podman machine initialization

### ISSUE THREAD

**Issue by odockal** [containers/podman-desktop#3196]
### Bug description

I do not see the image that was pulled successfully in Images page. This happens on fresh install of podman desktop and podman machine. Workaround is to force refresh (CTRL + R), then it does not happen again.

The same issue was spotted by @xbabalov on her pull image test PR - locally on linux. I have also ecnounter it on Linux, but was not able to reproduce, probably needs proper clean up to reproduce.

### Operating system

Windows 10 Pro, Mac OS 13 Intel, Fedora 37

### Installation Method

Installer from website/GitHub releases

### Version

1.2.0

### Steps to reproduce

0. No previous machine, images, containers are in the test env. 
1. Install of Podman Desktop
2. Installation of podman
3. Initializing of podman machine
4. Starting up the machine (default setup via dashboard)
5. Pull an image -> `quay.io/podman/hello`
6. Press done once it is complete
7. Assert: Image page is shown and there is the `hello` image present
Actual result: No image is present. Cli `podman images` shows the image.

Workaround: CTRL + R.

### Relevant log output

Nothing worth mentioning.
_No response_

### Additional context

This could be duplicate of #2806 or it is related.

**Comment by xbabalov** [comment]
The same issue appears on macOS, including not showing existing containers. Force refresh of Podman Desktop didn't help. 

![pd image container bug](https://github.com/containers/podman-desktop/assets/57829934/2b85bf74-d2ff-4249-bf3c-57725d42233a)

**Comment by benoitf** [comment]
Use the troubleshooting page where stores are listed ( bulb icon in status bar on the Right)

Then display images store and what are the events that have triggered the refresh

Then click on 'manual update' and see if it solves the issue

**Comment by benoitf** [comment]
also check what is displayed for containers in troubleshooting page

**Comment by xbabalov** [comment]
> Use the troubleshooting page where stores are listed ( bulb icon in status bar on the Right)
> 
> Then display images store and what are the events that have triggered the refresh
> 
> Then click on 'manual update' and see if it solves the issue

This helped. When I try pulling a new image afterwards it doesn't show until I manually refresh it again in the troubleshooting page. 

![images store updates](https://github.com/containers/podman-desktop/assets/57829934/3a6fb4d1-e96a-4e23-a2d8-47c313bd685c)
![container store updates](https://github.com/containers/podman-desktop/assets/57829934/04f62441-5a7b-4b39-a096-6c0cf1b7b584)

**Comment by benoitf** [comment]
can you check all extensions are working properly ? Like no exception in one of them ?
Because it looks like we ignore events as an update event has not been received
This event is sent after all extensions are started

**Comment by xbabalov** [comment]
I've only encountered this bug https://github.com/containers/podman-desktop/issues/2795 when trying to create a new Kind resource, but other than that all extensions seem to work correctly.

**Comment by odockal** [comment]
I am seeing error coming from lima on both win and linux. 
```
index.ts:1451 main ↪️ [lima] Could not find podman socket at /home/odockal/.lima/podman/sock/podman.sock
console.<computed> @ index.ts:1451
```

**Comment by benoitf** [comment]
this one is ok. 


Could you check by removing some extension ?
To see if when adding/removing an extension, it starts to fail.

**Comment by odockal** [comment]
Disabling Lima does not help, so Lima seems ok. Only option left (I tried all extensions, except podman) is that podman extension is to blame. 
From the console:
```
main ↪️ Error while fetching API groups: Error: No active cluster!
main ↪️ Error while fetching API groups: Error: No active cluster!
main ↪️ PluginSystem: received dom-ready event from the UI
main ↪️ Delayed startup, flushing 9 events
main ↪️ System ready. Loading extensions...
main ↪️ Activating extension (podman-desktop.compose)
main ↪️ Activating extension (podman-desktop.docker)
main ↪️ Activating extension (podman-desktop.kind)
main ↪️ Activating extension (podman-desktop.kube-context)
main ↪️ [kube-context] starting extension kube-context
main ↪️ Activating extension (podman-desktop.lima)
main ↪️ [lima] Could not find podman socket at C:\Users\Ondrej\.lima\podman\sock\podman.sock
main ↪️ Activating extension (podman-desktop.podman)
main ↪️ Activating extension (podman-desktop.registries)
main ↪️ Activating extension (podman-desktop.compose) ended
main ↪️ Activating extension (podman-desktop.docker) ended
main ↪️ Activating extension (podman-desktop.lima) ended
main ↪️ Activating extension (podman-desktop.kube-context) ended
main ↪️ Activating extension (podman-desktop.registries) ended
main ↪️ Fetched https://registry.podman-desktop.io/api/extensions.json in 214.1638000011444ms
main ↪️ Fetched https://registry.podman-desktop.io/api/extensions.json in 120.59979999996722ms
main ↪️ [kind] kind extension is active
main ↪️ Activating extension (podman-desktop.kind) ended
main ↪️ [kind] kind extension is active
main ↪️ Activating extension (podman-desktop.podman) ended
main ↪️ PluginSystem: initialization done.
main ↪️ Autostarting container engine
main ↪️ [podman] Podman extension: Do not start a machine as there is already one starting or started
main ↪️ [kind] kind extension is active
main ↪️ Fetched https://registry.podman-desktop.io/api/extensions.json in 112.30000000074506ms
main ↪️ (node:11532) UnhandledPromiseRejectionWarning: Error: EINVAL: invalid argument, stat '//./pipe/' (Use `Podman Desktop --trace-warnings ...` to show where the warning was created)
main ↪️ (node:11532) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 4)
main ↪️ (node:11532) UnhandledPromiseRejectionWarning: Error: EINVAL: invalid argument, stat '//./pipe/'
main ↪️ (node:11532) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 5)
```

**Comment by odockal** [comment]
After force refresh, the error is gone (before was repeatedly appearing) on windows. no more error coming from podman. And it works fine (image pull).

**Comment by odockal** [comment]
I would say that we can stick to Steps to reproduce. CTRL+R or force refresh works on all platforms. It is not restricted to rootful or rootless machine. 

What @xbabalov was fighting with was also connected to another issue: https://github.com/containers/podman-desktop/issues/3065 or https://github.com/containers/podman-desktop/issues/2582

**Comment by jeffmaury** [comment]
I reproduced on my test Win10 machine

**Comment by odockal** [comment]
@jeffmaury thx.

**Comment by jeffmaury** [comment]
According to the code, the image stores skip all events before the `extensions-already-started` event.
In my use case it is never received by the store.

![stores](https://github.com/containers/podman-desktop/assets/695993/16f06c09-f825-4b3b-af2a-3d13a42c64d9)

**Comment by odockal** [comment]
Same here. @benoitf was suggesting some of the extension to be blamed. I have reduced it to podman extension, but not sure how to overcome it. Seem comments above.

**Comment by benoitf** [comment]
https://github.com/containers/podman-desktop/blob/a7d0e05bb6654db9588efd27e212c1ee01a86fe6/packages/renderer/src/Loader.svelte#L27

This is where we dispatch the event

**Comment by jeffmaury** [comment]
Got it reproduced with the binary `window.extensionSystemIsExtensionsStarted()` returns false so event is not broadcasted

**Comment by jeffmaury** [comment]
I have a tentative PR, waiting for @benoitf comment, not fully tested but seems stores are properly initialized

**Comment by jeffmaury** [comment]
Tested on my Win11 Pro machine:

- launched the compiled bits
- install Podman
- init and start Podman machine
- pull an image
- image list not empty anymore

Seems fixed

**Comment by slemeur** [comment]
I confirm that the proposed fix, solved the issue on my side.

**Comment by odockal** [comment]
@jeffmaury It works for me on Fedora 37. I can easily see the issues fixed between version 1.2.0 and 0.13.0-next built locally. 
Although, there is a warning in the console log which could be a regression of this patch:
```
MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 extensions-started listeners added. Use emitter.setMaxListeners() to increase limit
    at _addListener (node:electron/js2c/sandbox_bundle:2:32989)
    at EventEmitter.addListener (node:electron/js2c/sandbox_bundle:2:35877)
    at Function.receive (<anonymous>:2:283)
```
